datasetIngestor/syncDataToFileserver_unix.go

```diff
@@ -21,27 +23,52 @@ func SyncDataToFileserver(datasetId string, user map[string]string, RSYNCServer
 	// append trailing slash to sourceFolder to indicate that the *contents* of the folder should be copied
 	// no special handling for blanks in sourceFolder needed here
 	fullSourceFolderPath := sourceFolder + "/"
-	// check if filelisting given
-	// rsync can create only one level deep directory structure, here we need more, therefore mkdir -p
-	// This code is no longer needed, sine Edgar has a new rrsync wrapper which craetes the needed directory
-	// cmd := exec.Command("/usr/bin/ssh",RSYNCServer,"mkdir","-p",destFolder)
-	// // show rsync's output
-	// cmd.Stdout = os.Stdout
-	// cmd.Stderr = os.Stderr
-	//
-	// fmt.Printf("Running %v.\n", cmd.Args)
-	// cmd.Run()
-
-	cmd := exec.Command("/usr/bin/rsync", "-e", "ssh -q", "-avxz", "--progress", "--msgs2stderr", fullSourceFolderPath, serverConnectString)
-	// // TODO: create folderstructure mkdir -p also for this case:
+	
+	versionNumber, err := getRsyncVersion()
+	if err != nil {
+		log.Fatal("Error getting rsync version: ", err)
+	}
+	
+	// Check rsync version and adjust command accordingly
+	var rsyncCmd *exec.Cmd
+	rsyncFlags := []string{"-e", "ssh", "-avxz", "--progress"}
 	if absFileListing != "" {
-		cmd = exec.Command("/usr/bin/rsync", "-e", "ssh -q", "-avxzr", "--progress", "--msgs2stderr", "--files-from", absFileListing, fullSourceFolderPath, serverConnectString)
+		rsyncFlags = append(rsyncFlags, "-r", "--files-from", absFileListing)
 	}
-	// show rsync's output
-	// cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
+	if version.Compare(versionNumber, "3.2.3", ">=") {
+		rsyncFlags = append(rsyncFlags, "--stderr=error")
+		rsyncCmd = exec.Command("/usr/bin/rsync", append(rsyncFlags, fullSourceFolderPath, serverConnectString)...)
```

Comment:
sorry, I missed this yesterday... since the rsyncCmd is the same in the `if` and `else` I think we can move it after the block.

If we do this, it will become easy (I guess) to do the rsyncFlags build in a function which can later be tested

Reply:
You didn't miss it yesterday, I just changed the code as I noticed a mistake in the code.

Reply:
It is now a separate function and fully tested

Reply:
thx!